chain/state/statetree.go

```diff
@@ -115,17 +117,18 @@ func (ss *stateSnaps) deleteActor(addr address.Address) {
 	ss.layers[len(ss.layers)-1].actors[addr] = streeOp{Delete: true}
 }
 
-func NewStateTree(cst cbor.IpldStore) (*StateTree, error) {
+func NewStateTree(cst cbor.IpldStore, version network.Version) (*StateTree, error) {
```

Comment:
In the future, we should _seriously_ consider encoding some form of _version_ number into the root node. This is one of the more painful parts of this patch.